subworkflows/local/phylogenetic_qc.nf

```diff
@@ -0,0 +1,49 @@
+//
+// Seqinspector Phylogenetic classification of reads, to check for contamination and adjacent issues
+//
+
+include { UNTAR as UNTAR_KRAKEN2_DB } from '../../modules/nf-core/untar/main.nf'
+include { KRAKEN2_KRAKEN2 } from '../../modules/nf-core/kraken2/kraken2/main.nf'
+include { KRONA_KTUPDATETAXONOMY } from '../../modules/nf-core/krona/ktupdatetaxonomy/main.nf'
+include { KRONA_KTIMPORTTAXONOMY } from '../../modules/nf-core/krona/ktimporttaxonomy/main.nf'
+
+workflow PHYLOGENETIC_QC{
+    take:
+    reads
+
+    main:
+    ch_reads = reads
+    ch_multiqc_files = Channel.empty()
+    ch_versions             = Channel.empty()
+    //
+    // MODULE: Untar kraken2_db
+    //
+    UNTAR_KRAKEN2_DB ( [ [:], params.kraken2_db ])
+    ch_kraken2_db = UNTAR_KRAKEN2_DB.out.untar.map { it[1] }
```

Comment:
I think it would be nice to be able to run this with an uncompressed database, I think nf-core/methylseq has something that checks if the file is compressed or not for instance: https://github.com/nf-core/methylseq/blob/d16f8f81e812d86f4d297e69de3f001628dfd5b4/subworkflows/local/prepare_genome/main.nf#L40